NanoGardener/python/modules/EFTReweighter.py

```diff
@@ -0,0 +1,248 @@
+
+import ROOT
+import math 
+import ctypes
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection 
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.modules.common.collectionMerger import collectionMerger
+
+import os.path
+
+class EFTReweighter(Module):
+    def __init__(self):
+        
+        self.cmssw_base = os.getenv('CMSSW_BASE')
+        self.cmssw_arch = os.getenv('SCRAM_ARCH')
+
+        ROOT.gSystem.AddIncludePath("-I"+self.cmssw_base+"/interface/")
+        ROOT.gSystem.AddIncludePath("-I"+self.cmssw_base+"/src/")
+        ROOT.gSystem.Load("libZZMatrixElementMELA.so")
+        ROOT.gSystem.Load(self.cmssw_base+"/src/ZZMatrixElement/MELA/data/"+self.cmssw_arch+"/libmcfm_707.so")
+
+        try:
+            ROOT.gROOT.LoadMacro(self.cmssw_base+'/src/LatinoAnalysis/Gardener/python/variables/melaHiggsEFT.C+g')
+        except RuntimeError:
+            ROOT.gROOT.LoadMacro(self.cmssw_base+'/src/LatinoAnalysis/Gardener/python/variables/melaHiggsEFT.C++g')
+      
+        self.mela = ROOT.Mela(13, 125,  ROOT.TVar.SILENT) 
+
+    def beginJob(self):
+        pass
+
+    def endJob(self):
+        pass
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+
+        filename = str(inputFile)[str(inputFile).find("nanoLatino"):str(inputFile).find(".root")+5]
+
+        if "_VBF_H0" in filename :
+          self.productionProcess = "VBF"
+        elif "_H0" in filename :
+          self.productionProcess = "GluGlu"
+        else:
+          raise NameError(filename, "is an unrecognised simulation")
```

Comment:
Again we might need to have another set of strings for Powheg samples here

Reply:
This module has been expanded to also deal with the ZH and WH AC samples. See reply above with respect to the powheg SM samples.